features/consent/src/main/java/in/projecteka/jataayu/consent/viewmodel/RequestedConsentListViewModel.kt

```diff
@@ -0,0 +1,106 @@
+package `in`.projecteka.jataayu.consent.viewmodel
+
+import `in`.projecteka.jataayu.consent.R
+import `in`.projecteka.jataayu.consent.extension.requestedConsentList
+import `in`.projecteka.jataayu.consent.model.ConsentFlow
+import `in`.projecteka.jataayu.consent.model.ConsentsListResponse
+import `in`.projecteka.jataayu.consent.repository.ConsentRepository
+import `in`.projecteka.jataayu.core.model.Consent
+import `in`.projecteka.jataayu.core.model.LinkedCareContext
+import `in`.projecteka.jataayu.core.model.Links
+import `in`.projecteka.jataayu.core.model.RequestStatus
+import `in`.projecteka.jataayu.core.model.RequestStatus.DENIED
+import `in`.projecteka.jataayu.core.model.RequestStatus.REQUESTED
+import `in`.projecteka.jataayu.core.model.grantedconsent.GrantedConsentDetailsResponse
+import `in`.projecteka.jataayu.core.model.grantedconsent.LinkedHip
+import `in`.projecteka.jataayu.network.utils.PayloadLiveData
+import `in`.projecteka.jataayu.network.utils.fetch
+import `in`.projecteka.jataayu.presentation.callback.IDataBindingModel
+import `in`.projecteka.jataayu.util.extension.EMPTY
+import `in`.projecteka.jataayu.util.ui.DateTimeUtils
+import android.content.res.Resources
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+
+class RequestedConsentListViewModel(private val repository: ConsentRepository) : ViewModel() {
```

Comment:
this can be renamed to RequestListViewModel (similar to what view this VM is related to)